lib/msf/core/post/file.rb

```diff
@@ -578,6 +579,32 @@ def copy_file(src_file, dst_file)
 
 protected
 
+  def _read_file_powershell(filename)
+    eof = cmd_exec("[System.IO.File]::ReadAllBytes(\"#{filename}\").Length").strip.to_i
+    data = ''
+    start = 0
+    chunk_size = 1024 * 1024
+    loop do
+      stop = start + chunk_size
+      stop = eof if stop > eof
+      data << _read_file_powershell_fragement(start, stop, filename)
+      start += chunk_size
+      break if start >= eof
+    end
+    return data
+  end
+
+  def _read_file_powershell_fragement(start, stop, filename)
+    b64_data= cmd_exec("$mstream = [System.IO.MemoryStream]::new();\
+      $gzipstream = [System.IO.Compression.GZipStream]::new($mstream, [System.IO.Compression.CompressionMode]::Compress);\
+      $get_bytes = [System.IO.File]::ReadAllBytes(\"#{filename}\")[#{start}..#{stop - 1}];\
+      $gzipstream.Write($get_bytes, 0 , $get_bytes.Length);\
+      $gzipstream.Close();\
+      [Convert]::ToBase64String($mstream.ToArray())")
+    uncompressed_fragement = Zlib::GzipReader.new(StringIO.new(Base64.decode64(b64_data))).read
+    return uncompressed_fragement
+  end
```

Comment:
"fragment" is misspelled in here, in both the method name and the `uncompressed_fragement` variable name.

Reply:
will fix this!